tapalcatl_server/storage.go

```diff
@@ -120,6 +120,13 @@ func (s *S3Storage) Fetch(t tapalcatl.TileCoord, c Condition) (*StorageResponse,
 	return result, nil
 }
 
+func (s *S3Storage) healthCheck() (error) {
+	key := "sanity_check.txt"
```

Comment:
> The configuration is parsed from JSON input using the storageDefinition and storageConfig structs. The convention seems to be that the storageConfig overrides the storageDefinition, for example when setting the key pattern.

I accept responsibility / blame for this definition / config notion :)
The idea is that in the future we'll have multiple s3 storage locations, with the only difference likely being the date prefix. I was going for an inheritance scheme where the "config" would override the "definition". And along the way I figured we might as well make more properties overridable, but perhaps that was a bad idea and is leading to too much duplication without any practical benefit. Really we just want a way to specify common properties once and mix those in somehow.

> While I'm sure there's a Better Way to Do It, the simplest thing for now would be to add another field alongside KeyPattern in those two structs, add docs and override the config, passing the value to the storage constructors where they can use it.

Yup, this is what I would do too.

> Should HealthCheck() error also be a part of the public Storage interface (L35-37)?

Agree here too. I'm surprised it worked without it, but maybe golang detected that all the storage implementations implemented this function?